Tasks/ANT/anttask.ts

```diff
@@ -208,63 +207,57 @@ async function doWork() {
-            var ccReportTask: string = null;
-            var reportBuildFile: string = null;
-            enableCodeCoverage();
-        }
-        else {
-            tl.debug("Option to enable code coverage was not selected and is being skipped.");
-        }
-
+        var summaryFile: string = null;
+        var reportDirectory: string = null;
+        var ccReportTask: string = null;
+        var reportBuildFile: string = null;
         var publishJUnitResults = tl.getInput('publishJUnitResults');
         var testResultsFiles = tl.getInput('testResultsFiles', true);
 
-        await antv.exec();
-
-        var buffer;
-        antb.on('stdout', (data) => {
-            if (data) {
-                buffer += data.toString();
-
-                let idx = buffer.indexOf(os.EOL);
-                while (idx > -1) {
-                    let line = buffer.substring(0, idx);
-
-                    processAntOutputLine(line);
+        //delete any previous cobertura instrumented classes as they might interfere with ant execution.
+        tl.rmRF(instrumentedClassesDirectory, true);
 
-                    buffer = buffer.substring(idx + os.EOL.length);
-                    idx = buffer.indexOf(os.EOL);
+        enableCodeCoverage().then(function (resp) {
+            ccReportTask = "CodeCoverage_9064e1d0";
+        }).catch(function (err) {
+            tl.warning("Failed to enable code coverage: " + err);
+        }).fin(function () {
+            
+            //antv.exec();
+            var buffer;
+            antb.on('stdout', (data) => {
+                if (data) {
+                    buffer += data.toString();
+                    let idx = buffer.indexOf(os.EOL);
+                    while (idx > -1) {
+                        let line = buffer.substring(0, idx);
+                        processAntOutputLine(line);
+                        buffer = buffer.substring(idx + os.EOL.length);
+                        idx = buffer.indexOf(os.EOL);
+                    }
                 }
-            }
-        });
-
-        antb.exec()
-            .then(function (code) {
-                publishTestResults(publishJUnitResults, testResultsFiles);
-                publishCodeCoverage(isCodeCoverageOpted);
-                tl.setResult(tl.TaskResult.Succeeded, "Task succeeded");
-            })
-            .fail(function (err) {
-                publishTestResults(publishJUnitResults, testResultsFiles);
-                console.error(err.message);
-                tl.debug('taskRunner fail');
-                tl.setResult(tl.TaskResult.Failed, err);
             });
+
+            antb.exec()
+                .then(function (code) {
+                    publishTestResults(publishJUnitResults, testResultsFiles);
+                    publishCodeCoverage(isCodeCoverageOpted);
+                    tl.setResult(tl.TaskResult.Succeeded, "Task succeeded");
+                })
+                .fail(function (err) {
```

Comment:
no reason for .fail here.  If you await above it will go into catch below and task will fail appropriately.